p2p/trust/trustmetric.go

```diff
@@ -181,11 +189,13 @@ func (tms *TrustMetricStore) saveToDB() {
 	peers := make(map[string]peerHistoryJSON, 0)
 
 	for key, tm := range tms.peerMetrics {
+		tm.mtx.Lock()
```

Comment:
isn't it better to grab lock outside the for loop once?

Reply:
the lock is specific to each metric tm refers to during the loop iterations

Reply:
Let's try to keep locks at the top of public methods. For instance here we can have `tm.HistoryJSON()` that handles the locking and returns the peerHistoryJSON